### PR TITLE
New Mexico Medical Care Expense Deduction

### LINKED ISSUE #2423: New Mexico Medical Care expense deduction
[Tax Form Instructions](https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf) page 26
[Legislative source](https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/856ebf4b-3814-49dd-8631-ebe579d6a42b/Personal%20Income%20Tax.pdf) page 233

Includes the Unreimbursed and uncompensated medical care expenses. Page 25 [l.s.](https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/856ebf4b-3814-49dd-8631-ebe579d6a42b/Personal%20Income%20Tax.pdf)

### PR DESCRIPTION
**Pull request by qingqiuzhang**
Fixes #2423 

### Summary
📄🧮💊

This pull request adds the New Mexico medical care expense deduction to the policy engine model for New Mexico state taxes. It includes YAML files for the deduction parameters for different filing statuses, a Python file for the deduction variable and formula, and a YAML file for a test case. The changes aim to improve the accuracy and completeness of the policy engine model for New Mexico state taxes.

> _`New Mexico` tax_
> _Adding medical deduction_
> _Files like autumn leaves_

### Walkthrough
*  Add the New Mexico medical care expense deduction to the policy engine model for New Mexico state taxes ([link](https://github.com/PolicyEngine/policyengine-us/pull/2476/files?diff=unified&w=0#diff-9c0f46950f45e6e8bba8b4e2a6f6fa6aea8c4b51ed2d2dfe6667ce329f36c205L1-R22), [link](https://github.com/PolicyEngine/policyengine-us/pull/2476/files?diff=unified&w=0#diff-9340b2bbb94665f4c4739ae58812885f4a99342a74d0808a8fd864240942a79fL1-R22), [link](https://github.com/PolicyEngine/policyengine-us/pull/2476/files?diff=unified&w=0#diff-635ae7ca2f27c1af6b718a031af3e80e47f1e056091b4256cc635ae48081c39eL1-R22), [link](https://github.com/PolicyEngine/policyengine-us/pull/2476/files?diff=unified&w=0#diff-226949622c77b7d13dec8ea9dc1326eff8b7101db5ca1e60e407fc5595fa28d1R1-R23), [link](https://github.com/PolicyEngine/policyengine-us/pull/2476/files?diff=unified&w=0#diff-ec2ded4a95a117659b1292d518dd349e7249b3e631ba09123b793af487f8c2eaL1-R22), [link](https://github.com/PolicyEngine/policyengine-us/pull/2476/files?diff=unified&w=0#diff-1dae6f4f9862c9debe1ecf3d343af505e1e2353c3bc1a6f7c44a50b41431d30fR1-R43))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+description: New Mexico medical care expense deduction at these rates for individuals filing as head of household, depending on the adjusted gross income.
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  rate_unit: /1
+  threshold_period: year
+  label: New Mexico medical care expense deduction rates for head of household filers.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml:7]:
```suggestion
  label: New Mexico medical care expense deduction head of household filers rate
```

File: policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  reference:
+    - title: 7-2-35. DEDUCTION - UNREIMBURSED OR UNCOMPENSATED MEDICAL CARE EXPENSES
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml:9]: add tax form reference

File: policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+description: New Mexico medical care expense deduction at these rates for individuals filing as head of household, depending on the adjusted gross income.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml:1]:
```suggestion
description: New Mexico provides the medical care expense deduction at these rates for individuals filing as head of household, depending on the adjusted gross income.
```

File: policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  rate_unit: /1
+  threshold_period: year
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml:6]: and other yaml files

File: policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+from policyengine_us.model_api import *
+import numpy as np
+
+
+class nm_mediacal_care_expense_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "NM medical care expense deduction"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py:8]:
```suggestion
    label = "New Mexico medical care expense deduction"
```

File: policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        rate = select(
+            [
+                filing_status == statuses.SINGLE,
+                filing_status == statuses.JOINT,
+                filing_status == statuses.HEAD_OF_HOUSEHOLD,
+                filing_status == statuses.SEPARATE,
+                filing_status == statuses.WIDOW,
+            ],
+            [
+                p.single.calc(agi, right=True),
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py:37]:
```suggestion
                p.single.calc(agi),
```

File: policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                p.joint.calc(agi, right=True),
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py:38]:
```suggestion
                p.joint.calc(agi),
```

File: policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+                p.head_of_household.calc(agi, right=True),
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py:39]:
```suggestion
                p.head_of_household.calc(agi),
```

File: policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+                p.separate.calc(agi, right=True),
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py:40]:
```suggestion
                p.separate.calc(agi),
```

File: policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+                p.widow.calc(agi, right=True),
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py:41]:
```suggestion
                p.widow.calc(agi),
```

File: policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        # the deduction amount is based on filing status, agi, and eligible expenses.
+        filing_status = tax_unit("filing_status", period)
+        statuses = filing_status.possible_values
+        agi = tax_unit("adjusted_gross_income", period)
+        expenses = add(
+            tax_unit,
+            period,
+            ["medical_out_of_pocket_expenses"],
+        )
+        # Use `right=True` to reflect "over ... but not over ...".
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py:27]: this is done by default through the brackets in the yaml files - the top bracket is inclusive and the bottom exclusive

Comment (qingqiuzhang) [reply policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py:27]: I've just tested without `right=TRUE`, cases on the cutoff points failed. It seems like the bottom bracket is inclusive and the other is exclusive.

Comment (qingqiuzhang) [reply policyengine_us/variables/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.py:27]: Sorry, I mean without `right=True`.

File: policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml [hunk 12]
```diff
@@ -0,0 +1,27 @@
+description: New Mexico provides the medical care expense deduction at these rates for individuals filing as head of household, depending on their adjusted gross income.
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  rate_unit: /1
+  label: New Mexico medical care expense deduction head of household filers rate
+  reference:
+    - title: 7-2-35. DEDUCTION - UNREIMBURSED OR UNCOMPENSATED MEDICAL CARE EXPENSES
+      href: https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/856ebf4b-3814-49dd-8631-ebe579d6a42b/Personal%20Income%20Tax.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml:9]: could you link the pages to the pdfs

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml:9]: all yaml files and references

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml:9]: also please add [this](https://nmonesource.com/nmos/nmsa/en/item/4340/index.do#!fragment/zoupio-_Toc140503888/BQCwhgziBcwMYgK4DsDWszIQewE4BUBTADwBdoAvbRABwEtsBaAfX2zgEYAWABgFYeAZgAcogJQAaZNlKEIARUSFcAT2gBydRIiEwuBIuVrN23fpABlPKQBCagEoBRADKOAagEEAcgGFHE0jAAI2hSdjExIA) as a source

File: policyengine_us/tests/policy/baseline/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.yaml [hunk 13]
```diff
@@ -0,0 +1,399 @@
+- name: 2015 single filer adjsuted gross income = 10_000 and medical out of pocket expense = 1_000, the deduction amount is 250
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.yaml:1]:
```suggestion
- name: 2015 single filer adjusted gross income = 10_000 and medical out of pocket expense = 1_000, the deduction amount is 250
```

Comment (PavelMakarchuk) [reply policyengine_us/tests/policy/baseline/gov/states/nm/tax/deductions/nm_medical_care_expense_deduction.yaml:1]: all tests

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml [hunk 14]
```diff
@@ -0,0 +1,29 @@
+description: New Mexico provides the medical care expense deduction at these rates for individuals filing as head of household, depending on their adjusted gross income.
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  rate_unit: /1
+  label: New Mexico medical care expense deduction head of household filers rate
+  reference:
+    - title: 7-2-35. DEDUCTION - UNREIMBURSED OR UNCOMPENSATED MEDICAL CARE EXPENSES Page 237
+      href: https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/856ebf4b-3814-49dd-8631-ebe579d6a42b/Personal%20Income%20Tax.pdf
+    - title: Tax Form Instructions Page PIT-1-26 LINE 16. Medical Care Expense Deduction
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nm/tax/income/deductions/medical_care_expense/head_of_household.yaml:10]:
```suggestion
    - title: Tax Form Instructions Page PIT-1-26 LINE 16. Medical Care Expense Deduction (page 31)
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**